### DISCUSSION TITLE
Trouble generating the Sentinel 2 Level 2 Product

### DISCUSSION THREAD

**Discussion by AndresMBID** [davidfrantz/force#333] [Q&A]
Hello, first of all I would like to thank you for the effort to make Force and keep the discussion forum active. 

This is my first time using Force and my problem is that I am not able to generate Sentinel level 2 products. I have tried both with a linux virtual machine and with docker on windows (latest version) and the result is the same. I got my S2 data from Google Cloud Storage.

The processing seems to run, and for a long time it is active but finally there is always a killed process warning and the processing is terminated. The whole list of images to process goes from QUEUED to DONE and the logs generated are empty and in the level2 folder no file is generated.

To simplify the process, I have tested only with a sentinel 2 image and to avoid memory problems I have set the number of processes to 1 and the number of threads to 1 as well. And I also reduced the tile size and batch size. And I get the same problem. I have also changed dem, one of 10m resolution, another one of 30m, different projections.... 

I leave you my parameter file, the capture of how the memory usage behaves during processing, the output log and another capture of the command terminal. I don't know what else to do or what parameter to change. 

[l2ps.txt](https://github.com/user-attachments/files/17714801/l2ps.txt)
![docker_stats](https://github.com/user-attachments/assets/868c8497-d04b-476c-a465-9e7e0016818a)
[S2A_MSIL1C_20220820T113331_N0400_R080_T29TNH_20220820T150918.SAFE.log](https://github.com/user-attachments/files/17714944/S2A_MSIL1C_20220820T113331_N0400_R080_T29TNH_20220820T150918.SAFE.log)
![command_terminal_screenshot](https://github.com/user-attachments/assets/a3bd39e3-9b39-495b-babd-1dcdce6c1e7b)

Thank you for your answers.

**Comment by AndresMBID** [comment]
Finally I had to increase the memory used by docker to process the sentinel image. I did this by generating a wsl configuration file in the following path in windows “C:\UsersUser.wslconfig” following the indications of https://learn.microsoft.com/en-us/windows/wsl/wsl-config#wslconfig. Then restart wsl and regenerate the container. 

"C:\Users\User\.wslconfig"
```
	# Settings apply across all Linux distros running on WSL 2
	[wsl2]

	# Limits VM memory to use no more than 4 GB, this can be set as whole numbers using GB or MB
	memory=10GB 
```

In powersell I ran this
```
wsl --shutdown #restart wsl
docker run --memory=10g --memory-swap=10g -d -it -v D:/201_Force:/opt/data --name force davidfrantz/force /bin/bash
```



However, I got the following error. Do you know why?
```
/opt/data/level1/T29TNH/S2A_MSIL1C_20220820T113331_N0400_R080_T29TNH_20220820T150918.SAFE
: dc:  99.79%. wc:   5.42%. sc:   0.11%. cc:   8.47%. AOD: 0.0965. # of targets: 216/18805. Error while reading from tile file!
Reading tile file failed! Tiling images failed! Error in geometric module.
```

**Comment by davidfrantz** [comment]
Hi @AndresMBID,

yes, "killed" messages are almost always related to memory.

The current error is because of this line in your parameter file, which seems to hold output options, but not a tile allow list:

``
FILE_TILE = /opt/data/log/Output_options
``

Either disable the usage of the file, or provide a proper tile allow list.

Best,
David

**Reply by AndresMBID** [reply]
Thank you! it was just that. 

Best,
Andrés.
